Source/Tools/LinuxEmulation/LinuxSyscalls/ThreadManager.h

```diff
@@ -22,6 +22,80 @@ namespace FEX::HLE {
 class SyscallHandler;
 class SignalDelegator;
 
+// This is similar to FEXCore::InterruptableConditionVariable with the exception that notifying only occurs if there is a waiter and can check if
```

Comment:
Comparison to other synchronization primitives should be a side note, not the main introduction to a class. The docstring should describe the purpose/use of the class directly rather than approximating it indirectly (as it currently does).